unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/AvatarFactory.cs

```diff
@@ -0,0 +1,54 @@
+using DCL;
+using GPUSkinning;
+using UnityEngine;
+
+namespace AvatarSystem
+{
+    public class AvatarFactory : IAvatarFactory
+    {
+        private readonly ServiceLocator serviceLocator;
+
+        public AvatarFactory(ServiceLocator serviceLocator)
+        {
+            this.serviceLocator = serviceLocator;
+        }
+
+        public Avatar CreateAvatar(GameObject avatarContainer, IAnimator animator, ILOD lod, IVisibility visibility) =>
+            new (
+                CreateAvatarCurator(),
+                CreateLoader(avatarContainer),
+                animator,
+                visibility,
+                lod,
+                new SimpleGPUSkinning(),
+                new GPUSkinningThrottler(),
+                new EmoteAnimationEquipper(animator, DataStore.i.emotes)
+            );
+
+        public AvatarWithHologram CreateAvatarWithHologram(
+            GameObject avatarContainer,
+            Transform avatarRevealContainer,
+            GameObject armatureContainer,
+            IAnimator animator,
+            ILOD lod,
+            IVisibility visibility
+        ) =>
+            new (
+                new BaseAvatar(avatarRevealContainer, armatureContainer, lod),
+                CreateAvatarCurator(),
+                CreateLoader(avatarContainer),
+                animator,
+                visibility,
+                lod,
+                new SimpleGPUSkinning(),
+                new GPUSkinningThrottler(),
+                new EmoteAnimationEquipper(animator, DataStore.i.emotes)
+            );
+
+        private Loader CreateLoader(GameObject avatarContainer) =>
+            new (new WearableLoaderFactory(), avatarContainer, new AvatarMeshCombinerHelper());
+
+        private AvatarCurator CreateAvatarCurator() =>
+            new (new WearableItemResolver(), serviceLocator.Get<IEmotesCatalogService>());
```

Comment:
I think it would be not quite correct as we would enforce `ServiceLocatorFactory` to resolve dependencies instead of `ServiceLocator`.
Then we might be tempted to do so in every other case that would lead to an exponential growth of that factory.

Reply:
From one point, current approach breaks encapsulation a bit (because of the dependency on the big Container, who knows every single service that this class doesn't need). I also see, that the only one user of the `ServiceLocator.Register` method is a `ServiceLoactorFactory`,  our main point where we resolve dependency graph (manually), and it is totally fine. The only downside would be if we Rebind new service to `IEmotesCatalogService`, where  your approach would make fully sense. 
I would go with YAGNI, and add stick to the constructor injection of `IEmotesCatalogService`, until the case we have the case of Rebinding this Service. Of course, such approach could be not so "future-proof" (but I hope we will move to normal DI framework approach in future :) )

Reply:
I would like to know @AjimenezDCL opinion on the topic. Also, my suggestion was optional (that is why PR was 🟢)